src/binary/non_blocking/binary_buffer.rs

```diff
@@ -689,7 +690,7 @@ mod tests {
         let mut buffer = BinaryBuffer::new(&[0b1000_0000]);
         let var_int = buffer.read_uint(buffer.remaining())?;
         assert_eq!(var_int.size_in_bytes(), 1);
-        assert_eq!(var_int.value(), &UInt::U64(128));
+        assert_eq!(var_int.value(), &UInt::from(128));
```

Comment:
🗺️ I used `TypeName::from` instead of `value.into()` in several places because that's the only way to specify the expected type inline. You can't use turbofish syntax with `Into`.